index.bs

```diff
@@ -11302,6 +11306,153 @@ The [=remote end steps=] given |session| and |command parameters| are:
 </div>
 
 
+## The extensions Module ## {#module-extensions}
+
+The <dfn export for=modules>extensions</dfn> module contains functionality for
+managing and interacting with extensions.
+
+### Definition ### {#module-extensions-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+ExtensionsCommand = (
+  extensions.Install
+)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+ExtensionsResult = (
+  extensions.InstallResult
+)
+</pre>
+
+### Types ### {#module-extensions-types}
+
+#### The extensions.Extension Type #### {#type-extensions-Extension}
+
+<pre class="cddl local-cddl">
+extensions.ExtensionArchive = {
+   files: [+text],
+}
+</pre>
+
+<pre class="cddl remote-cddl local-cddl">
+extensions.Extension = {
+   installPath: extensions.ExtensionPath / extensions.ExtensionArchivePath / extensions.ExtensionBase64Encoded,
```

Comment:
In Chrome, we currently only allow installing extensions if the client is local to the browser (i.e., connected via pipes instead of the WebSocket). We could probably relax this based on a command line argument but I wonder if it makes sense to start with the extensions.ExtensionPath only?

cc @oliverdunk for additional thoughts

Reply:
Could that be managed by the webdriver? Archives (zip/crx) and Base64 strings are can effectively be unpacked to a temporary folder, and then run as from any other path.

Reply:
the driver process and the browser might not be running on the same machine so, in a general case, we probably should not handle it in the driver. I think a remote-end path to an archive or to an unpacked folder is less of a concern but installing an extension from base64 would mean writing something to the filesystem that was not present there originally.

Reply:
@OrKoN, I'm fairly certain Selenium currently supports loading an extension from base64 in WebDriver. Do you have any idea how that was implemented?

Reply:
Extensions can be quite big. I think AdblockPlus is around 60mb while Ghostery is around 12mb.